client/src/main/java/io/pravega/client/stream/impl/ClientFactoryImpl.java

```diff
@@ -173,7 +175,13 @@ public ClientFactoryImpl(String scope, Controller controller, ConnectionFactory
                 synchronizerConfig);
         ReaderGroupStateManager stateManager = new ReaderGroupStateManager(readerId, sync, controller, nanoTime);
         stateManager.initializeReader(config.getInitialAllocationDelay());
-        return new EventStreamReaderImpl<T>(inFactory, metaFactory, s, stateManager, new Orderer(), milliTime, config);
+        Builder<Stream, WatermarkReaderImpl> watermarkReaders = ImmutableMap.builder();
+        if (!config.isDisableTimeWindows()) {
+            for (Stream stream : stateManager.getStreams()) {
+                watermarkReaders.put(stream, new WatermarkReaderImpl(stream, this, connectionFactory.getInternalExecutor()));
```

Comment:
We can create `RevisionedStreamClient` here and pass it to the WatermarkReader here.

Reply:
Done